### PR TITLE
292294 doggy bail

### PR DESCRIPTION
**Pull request by jbarnsley10**
Allow CDO to be created as 'interim scheme' with new filed 'applicationType'
Extended Snyk exclusion for inflight
Added story 306380

### CODE REVIEW

File: app/register/search-basic.js [hunk 1]
```diff
@@ -90,6 +93,17 @@ const sortOwnerSearch = (a, b) => {
   return a.distance - b.distance || b.rank - a.rank || `${a.lastName} ${a.firstName}` - `${b.lastName} ${b.firstName}`
 }
 
+const addFullDogIndexIfMissing = arr => {
```
Comment (steve-dickinson) [inline app/register/search-basic.js:96]: You could use map:

```
const addFullDogIndexIfMissing = arr => {
  return arr.map(elem => {
    return (elem.length === 6 && /^\d+$/.test(elem)) ? `ed${elem}` : elem;
  });
}
```

File: app/repos/dogs.js [hunk 2]
```diff
@@ -149,6 +152,16 @@ const getStatusId = (statuses, statusName) => {
   return statuses.filter(x => x.status === statusName)[0].id
 }
 
+const determineStartingStatus = (dog, statuses) => {
```
Comment (steve-dickinson) [inline app/repos/dogs.js:155]: You could use a tenery ooperator to make it more concise:

```
const determineStartingStatus = (dog, statuses) => {
  return dog.source === 'ROBOT' 
    ? getStatusId(statuses, constants.statuses.Exempt)
    : dog.interimExemption 
      ? getStatusId(statuses, constants.statuses.InterimExempt)
      : getStatusId(statuses, constants.statuses.PreExempt);
}
```

**Review by steve-dickinson — COMMENTED** [review]
I've left a couple of comments around potencial refactoring to tidy things up a bit and make things a little more manageable.

File: app/register/search-basic.js
[the remark this answers is not in the document]
Comment (jbarnsley10) [reply app/register/search-basic.js:1]: Good suggestion

File: app/repos/exemption.js
Comment (steve-dickinson) [inline app/repos/exemption.js:1]: This file needs a little bit of work especially the updateExemption. It has a lot of nesting and complexity in the function. Look at refactoring and breaking it down into smaller more manageable functions.

**Review by steve-dickinson — COMMENTED** [review]
Refactoring work required in the app > repos > exemption.js file

**Review by steve-dickinson — APPROVED** [review]
LGTM

**Pull request merged**